components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -284,6 +285,14 @@ public boolean validateScope(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                 }
             }
         }
+
+        String[] requestedOIDCScopes = null;
+        try {
+            requestedOIDCScopes = OAuth2Util.getRequestedOIDCScopes(tokReqMsgCtx.getScope());
```

Comment:
Can't we get the requested scopes from the request object?

Reply:
It won't have the scopes.